### PR TITLE
Add 1st Party CloudSQL Dashboards and Alerts

### PR DESCRIPTION
**Pull request by Mrod1598**
Add alerts and dashboards for CloudSQL. The dashboards contain alert charts that are dependent on the alerts in this PR as well.

### CODE REVIEW

File: alerts/cloudsql/README.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    }
+```
+
+#### Notification Channels
+
+The ID of the notification channel to be notified.
+
+```json
+    "notificationChannels": [
+        "projects/project-id/notificationChannels/1234567
```
Comment (antonblock) [inline alerts/cloudsql/README.md:37]:
```suggestion
        "projects/project-id/notificationChannels/1234567"
```

File: dashboards/cloudsql/1p-cloudsql-general.json [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+        "height": 4,
+        "widget": {
+          "alertChart": {
+            "name": "<insert alert name here or remove chart>"
+          }
+        },
+        "width": 4,
+        "xPos": 0,
+        "yPos": 14
+      },
```
Comment (EvanSimpson) [inline dashboards/cloudsql/1p-cloudsql-general.json:144]: We can't currently support alert charts in sample dashboards in a way that is useful to users, so let's remove these for now. The charts depend on the existence of a project-specific alert instance, and so will appear as an empty chart when imported.

File: dashboards/cloudsql/1p-cloudsql-general.json [hunk 3]
```diff
@@ -0,0 +1,320 @@
...
+        },
+        "width": 6,
+        "xPos": 0,
+        "yPos": 0
+      },
+      {
+        "height": 2,
+        "widget": {
+          "text": {
+            "content": "The below chart gives a count if how many replications are in each state.",
```
Comment (EvanSimpson) [inline dashboards/cloudsql/1p-cloudsql-general.json:231]:
```suggestion
            "content": "The below chart gives a count of how many replications are in each state.",
```

File: dashboards/cloudsql/1p-cloudsql-general.json [hunk 4]
```diff
@@ -0,0 +1,320 @@
...
+        },
+        "width": 4,
+        "xPos": 8,
+        "yPos": 14
+      },
+      {
+        "height": 2,
+        "widget": {
+          "text": {
+            "content": "The below chart gives a count if how many CloudSQL instances are in each state. ",
```
Comment (EvanSimpson) [inline dashboards/cloudsql/1p-cloudsql-general.json:218]:
```suggestion
            "content": "The below chart gives a count of how many CloudSQL instances are in each state. ",
```

File: dashboards/cloudsql/1p-cloudsql-general.json [hunk 5]
```diff
@@ -0,0 +1,320 @@
...
+        },
+        "width": 6,
+        "xPos": 6,
+        "yPos": 0
+      },
+      {
+        "height": 2,
+        "widget": {
+          "text": {
+            "content": "This logs widget shows the logs of when a backup has an error and can't back up.",
```
Comment (EvanSimpson) [inline dashboards/cloudsql/1p-cloudsql-general.json:244]:
```suggestion
            "content": "This logs widget shows the logs from when a backup has an error and can't back up.",
```

File: dashboards/cloudsql/1p-cloudsql-general.json [hunk 6]
```diff
@@ -0,0 +1,320 @@
...
+        },
+        "width": 4,
+        "xPos": 4,
+        "yPos": 6
+      },
+      {
+        "height": 2,
+        "widget": {
+          "text": {
+            "content": "This logs widget shows all the logs relating to cloudSQL database instances at the error severity level.",
```
Comment (EvanSimpson) [inline dashboards/cloudsql/1p-cloudsql-general.json:270]:
```suggestion
            "content": "This logs widget shows all the logs relating to CloudSQL database instances at the error severity level.",
```

File: dashboards/cloudsql/1p-cloudsql-general.json [hunk 7]
```diff
@@ -0,0 +1,320 @@
...
+          "text": {
+            "content": "This Alert chart fires when any instance enters a failed state.",
+            "format": "RAW"
+          },
+          "title": "CloudSQL - Instance in failed state"
+        },
+        "width": 4,
+        "xPos": 0,
+        "yPos": 12
+      },
```
Comment (EvanSimpson) [inline dashboards/cloudsql/1p-cloudsql-general.json:291]: Please remove

File: dashboards/cloudsql/1p-cloudsql-replication.json [hunk 8]
```diff
@@ -0,0 +1,267 @@
...
+          "height": 4,
+          "widget": {
+            "alertChart": {
+              "name": "<insert alert name here or remove chart>"
+            }
+          },
+          "width": 6,
+          "xPos": 6,
+          "yPos": 2
+        },
```
Comment (EvanSimpson) [inline dashboards/cloudsql/1p-cloudsql-replication.json:74]: Please remove alert chart

File: dashboards/cloudsql/1p-cloudsql-replication.json [hunk 9]
```diff
@@ -0,0 +1,267 @@
...
+            "text": {
+              "content": "Alert chart that fires when a replication enters an error state.",
+              "format": "RAW"
+            },
+            "title": "Replication In Error State"
+          },
+          "width": 6,
+          "xPos": 6,
+          "yPos": 0
+        },
```
Comment (EvanSimpson) [inline dashboards/cloudsql/1p-cloudsql-replication.json:212]: Please remove

File: dashboards/cloudsql/1p-cloudsql-transactions.json [hunk 10]
```diff
@@ -0,0 +1,333 @@
...
+          "height": 4,
+          "widget": {
+            "alertChart": {
+              "name": "projects/otel-agent-dev/alertPolicies/9123919143766465969"
+            }
+          },
+          "width": 4,
+          "xPos": 0,
+          "yPos": 14
+        },
```
Comment (EvanSimpson) [inline dashboards/cloudsql/1p-cloudsql-transactions.json:30]: Please remove alert chart

File: dashboards/cloudsql/1p-cloudsql-transactions.json [hunk 11]
```diff
@@ -0,0 +1,333 @@
...
+            "text": {
+              "content": "This Alert chart detects when a transaction has not been vacuumed before 10 additional transactions have happened",
+              "format": "RAW"
+            },
+            "title": "PostgreSQL - Slow Transactions"
+          },
+          "width": 4,
+          "xPos": 0,
+          "yPos": 12
+        },
```
Comment (EvanSimpson) [inline dashboards/cloudsql/1p-cloudsql-transactions.json:291]: Please remove

**Review by EvanSimpson — CHANGES_REQUESTED** [review]
Can we also change the `cloudsql` folders to `google-cloudsql`? Thanks.

File: dashboards/cloudsql/1p-cloudsql-transactions.json (hunk 11 above)
Comment (Mrod1598) [reply dashboards/cloudsql/1p-cloudsql-transactions.json]: removed

File: dashboards/cloudsql/1p-cloudsql-replication.json (hunk 8 above)
Comment (Mrod1598) [reply dashboards/cloudsql/1p-cloudsql-replication.json]: removed

File: dashboards/cloudsql/1p-cloudsql-transactions.json (hunk 10 above)
Comment (Mrod1598) [reply dashboards/cloudsql/1p-cloudsql-transactions.json]: removed

File: dashboards/cloudsql/1p-cloudsql-replication.json (hunk 9 above)
Comment (Mrod1598) [reply dashboards/cloudsql/1p-cloudsql-replication.json]: removed

File: dashboards/cloudsql/1p-cloudsql-general.json (hunk 7 above)
Comment (Mrod1598) [reply dashboards/cloudsql/1p-cloudsql-general.json]: removed

File: dashboards/cloudsql/1p-cloudsql-general.json (hunk 2 above)
Comment (Mrod1598) [reply dashboards/cloudsql/1p-cloudsql-general.json]: removed them for now.

File: dashboards/cloudsql/1p-cloudsql-general.json (hunk 3 above)
Comment (Mrod1598) [reply dashboards/cloudsql/1p-cloudsql-general.json]: fixed

File: dashboards/cloudsql/1p-cloudsql-general.json (hunk 4 above)
Comment (Mrod1598) [reply dashboards/cloudsql/1p-cloudsql-general.json]: fixed

File: dashboards/cloudsql/1p-cloudsql-general.json (hunk 5 above)
Comment (Mrod1598) [reply dashboards/cloudsql/1p-cloudsql-general.json]: fixed

File: dashboards/cloudsql/1p-cloudsql-general.json (hunk 6 above)
Comment (Mrod1598) [reply dashboards/cloudsql/1p-cloudsql-general.json]: fixed

File: dashboards/cloudsql/README.md [hunk 12]
```diff
@@ -0,0 +1,21 @@
+### Dashboards for CloudSQL
+
+#### Notes
+
+- These dashboards are based on first party Google Cloud monitoring solutions.
+- each of these dashboards have an alert associated with them that should be created, and the name added to the alert chart on the corresponding dashboard.
```
Comment (EvanSimpson) [inline dashboards/cloudsql/README.md:6]: We can remove this for now as well.

**Review by EvanSimpson — CHANGES_REQUESTED** [review]

File: dashboards/cloudsql/1p-cloudsql-general.json (hunk 2 above)
Comment (EvanSimpson) [reply dashboards/cloudsql/1p-cloudsql-general.json]: Thank you!

File: alerts/cloudsql/1p-postgresql-slow-transactions.json [hunk 13]
```diff
@@ -0,0 +1,34 @@
+{
+  "displayName": "PostgreSQL - Slow Transactions",
+  "documentation": {
+    "content": "This Alert fires if the database is having trouble keeping up transactions. improvements in performance should be investigated. those improvements could be increasing resources for cloudsql or improving query efficiency, for more information visit https://cloud.google.com/sql/docs/postgres/diagnose-issues.",
```
Comment (adeychman) [inline alerts/cloudsql/1p-postgresql-slow-transactions.json:4]: A small adjustment for language and capitalization of sentences: 

"This Alert fires if the database is having trouble keeping up transactions. You should investigate performance improvements, such as increasing resources for CloudSQL, improving query efficiency, and more. For additional information visit https://cloud.google.com/sql/docs/postgres/diagnose-issues.

File: alerts/cloudsql/1p-replication-in-error-state.json [hunk 14]
```diff
@@ -0,0 +1,34 @@
+{
+  "displayName": "Replication in Error State",
+  "documentation": {
+    "content": "This Alert fires when any replica has stopped working and has entered an error state. The cause should be investigated. for more information around disaster recovery visit: https://cloud.google.com/sql/docs/mysql/replication/cross-region-replicas",
```
Comment (adeychman) [inline alerts/cloudsql/1p-replication-in-error-state.json:4]: "This Alert fires when any replica has stopped working and has entered an error state.  For more information around disaster recovery visit: https://cloud.google.com/sql/docs/mysql/replication/cross-region-replicas"

**Review by adeychman — COMMENTED** [review]
A couple of small grammar sugestions

File: alerts/cloudsql/1p-postgresql-slow-transactions.json (hunk 13 above)
Comment (Mrod1598) [reply alerts/cloudsql/1p-postgresql-slow-transactions.json]: updated

File: alerts/cloudsql/1p-replication-in-error-state.json (hunk 14 above)
Comment (Mrod1598) [reply alerts/cloudsql/1p-replication-in-error-state.json]: updated

File: dashboards/cloudsql/README.md (hunk 12 above)
Comment (Mrod1598) [reply dashboards/cloudsql/README.md]: Removed.

File: dashboards/cloudsql/1p-cloudsql-general.json (hunk 2 above)
Comment (joyw-2018) [reply dashboards/cloudsql/1p-cloudsql-general.json]: Thank you. sorry for chime in late. Please remove the alert chart as we don't yet have support for it in sample dash.

**Review by EvanSimpson — APPROVED** [review]

**Comment by adeychman** [comment]
lgtm, thanks! It can be merged in

**Pull request merged**